### PR TITLE
[notify] Simplify Slack notifications code

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Following changes to the notifications behavior (removal of team-level channel notifications) and addition of the gitlab Python package, and before making further modifications to the task's behavior, I took the opportunity to remove some dead code, remove use of environment variables, and add utility functions.

### Motivation

Simplify the notification code, removing unneeded sections.
Avoid relying on environment variables, make use of the info given by the Gitlab API as much as possible.

### Additional Notes

Some implementation notes:
- I removed the exception handling code in `tasks/notify.py` because I don't think it does anything except re-print the traceback? If I'm wrong, I can add it back in
- I introduced the `get_pipeline` and `get_commit` utilities to get more kinds of objects from the Gitlab API, which were needed to remove the use of environment variables
- I then changed `get_failed_jobs` to take the result of `get_pipeline` as argument, to limit the number of calls to the API
- I renamed `CHANNEL_PIPELINES` to `PIPELINES_CHANNEL` for consistency with the rest of variable names (like `ALL_TEAMS_CHANNEL`)

Future changes:
- Pass the pipeline id as a proper parameter instead of an environment variable (I didn't want to change task signatures here)
- Target a different Slack channel on non-main deploy pipelines (I didn't want to change the behavior of the task here)
- Find the pipeline type without having to pass it as an explicit parameter

### Describe how to test/QA your changes

Ran the following within the container used in CI, with various cases (failing `main`, successful `7.56.x`, failing dev branch deploy pipeline with failed tests):
```
$ CI_PIPELINE_ID=39628099 invoke -e notify.send-message --notification-type "merge" --dry-run
Would send to #datadog-agent-pipelines:
:host-red: :merged: datadog-agent merge pipeline <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/39628099|39628099> for main failed.
Fix the admission controller on old versions of Kubernetes (<https://github.com/DataDog/DataDog/datadog-agent/pull/27600|#27600>) (<https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/6e5a91dbd42dc27c6ad882d7933e3d29ba44db77|ba44db77>)(:github: <https://github.com/DataDog/DataDog/datadog-agent/commit/6e5a91dbd42dc27c6ad882d7933e3d29ba44db77|link>) by Lénaïc Huard
Failed jobs:
- <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/577972220|new-e2e-installer> (`e2e` stage)
- <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/577972176|trigger_auto_staging_release> (`trigger_release` stage)
```

```
$ CI_PIPELINE_ID=39629659  invoke -e notify.send-message --notification-type "merge" --dry-run
Would send to #datadog-agent-pipelines:
:host-green: :merged: datadog-agent merge pipeline <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/39629659|39629659> for 7.56.x succeeded.
bump omnibus software commit to include cURL update (<https://github.com/DataDog/DataDog/datadog-agent/pull/27713|#27713>) (<https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/27a6961d79f5aaba05bb553ee50df23ff0bea5b5|f0bea5b5>)(:github: <https://github.com/DataDog/DataDog/datadog-agent/commit/27a6961d79f5aaba05bb553ee50df23ff0bea5b5|link>) by Hugo Beauzée-Luyssen
```

```
$ CI_PIPELINE_ID=39528120 invoke -e notify.send-message --notification-type "deploy" --dry-run
Would send to #datadog-agent-pipelines:
:host-red: :rocket: datadog-agent deploy pipeline <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/39528120|39528120> for test-revert-metadata-dup-v3 failed.
don't set annotations and labels on kubemetadata entities for pods and deployments (<https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/ac669ad51b5af44e1310b9af914f8d859c5431e9|9c5431e9>)(:github: <https://github.com/DataDog/DataDog/datadog-agent/commit/ac669ad51b5af44e1310b9af914f8d859c5431e9|link>) by adel121
Failed jobs:
- <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880832|tests_rpm-arm64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880819|tests_deb-arm64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880812|tests_rpm-x64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880785|tests_deb-x64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880776|prepare_sysprobe_ebpf_functional_tests_x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880775|prepare_sysprobe_ebpf_functional_tests_arm64> (`source_test` stage)
Failed tests:
- `TestDeploymentParser_Parse` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `Test_DeploymentsFakeKubernetesClient` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `TestPodParser_Parse` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `Test_PodsFakeKubernetesClient` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)

Would send to adel.hajhassan@datadoghq.com:
:host-red: :rocket: datadog-agent deploy pipeline <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/pipelines/39528120|39528120> for test-revert-metadata-dup-v3 failed.
don't set annotations and labels on kubemetadata entities for pods and deployments (<https://gitlab.ddbuild.io/DataDog/datadog-agent/-/commit/ac669ad51b5af44e1310b9af914f8d859c5431e9|9c5431e9>)(:github: <https://github.com/DataDog/DataDog/datadog-agent/commit/ac669ad51b5af44e1310b9af914f8d859c5431e9|link>) by adel121
Failed jobs:
- <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880832|tests_rpm-arm64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880819|tests_deb-arm64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880812|tests_rpm-x64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880785|tests_deb-x64-py3>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880776|prepare_sysprobe_ebpf_functional_tests_x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880775|prepare_sysprobe_ebpf_functional_tests_arm64> (`source_test` stage)
Failed tests:
- `TestDeploymentParser_Parse` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `Test_DeploymentsFakeKubernetesClient` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `TestPodParser_Parse` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
- `Test_PodsFakeKubernetesClient` from package `comp/core/workloadmeta/collectors/internal/kubeapiserver` (in <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880862|tests_windows-x64>, <https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/576880842|tests_macos> and 4 more)
```

### CODE REVIEW

File: tasks/libs/pipeline/data.py [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+    return repo.commits.get(git_ref)
+
+
+def get_pipeline(project_name: str, pipeline_id: str) -> ProjectPipeline:
+    """
+    Retrieves the git ref for a given pipeline id in a given project.
     """
     repo = get_gitlab_repo(project_name)
-    pipeline = repo.pipelines.get(pipeline_id)
+    return repo.pipelines.get(pipeline_id)
```
Comment (CelianR) [inline tasks/libs/pipeline/data.py:23]: Should this be within `gitlab_api.py` ?

**Review by CelianR — APPROVED** [review]
Amazing 👏 🎉

File: tasks/libs/pipeline/data.py (hunk 1 above)
Comment (KSerrania) [reply tasks/libs/pipeline/data.py:23]: Sounds good, let me move these two methods.

**Review by FlorentClarret — APPROVED** [review]

**Pull request merged**